### DISCUSSION TITLE
Can improve performance with large json file?

### DISCUSSION THREAD

**Discussion by seasoncool** [grafana/grafana-infinity-datasource#544] [Q&A] [answered]
We have single JSON file around 50mb as datasource of grafana, however , when we refresh the panle and it takes more than 10s to get results from JSON file.
Can we improve the performance about it ? or any ideas are welcomed. thanks in advance.

**Comment by yesoreyeram** [comment]
What does your query looks like? Sample json also helpful.. how long it takes to download? Is browser slow to render? Network slow? What parser you use...

But definitely this tool is not designed for use cases such as 50mb..

**Comment by seasoncool** [comment, marked as answer]
thanks for your response, we split each file finally. thanks.
